tests/poller.cpp

```diff
@@ -46,9 +41,7 @@ TEST(poller, move_construct_non_empty)
     a->add(socket, ZMQ_POLLIN, [](short) {});
     zmq::poller_t b = std::move(*a);
 
-    ASSERT_EQ(0u, a->size ());
     a.reset ();
-    ASSERT_EQ(1u, b.size ());
```

Comment:
Something must be asserted here. If there are no query operations, wait should be called on both pollers to ensure that only b has an effect.